types/hapi/definitions/response/response-toolkit.d.ts

```diff
@@ -0,0 +1,133 @@
+import {ResponseObject, ServerRealm, ServerStateCookieOptions} from "hapi";
+
+/**
+ * See more about Lifecycle
+ * https://github.com/hapijs/hapi/blob/master/API.md#request-lifecycle
+ *
+ */
+
+
+/**
+ * The response toolkit is a collection of properties and utilities passed to every [lifecycle method](https://github.com/hapijs/hapi/blob/master/API.md#lifecycle-methods)
+ * It is somewhat hard to define as it provides both utilities for manipulating responses as well as other information. Since the
+ * toolkit is passed as a function argument, developers can name it whatever they want. For the purpose of this
+ * document the h notation is used. It is named in the spirit of the RethinkDB r method, with h for hapi.
+ * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#response-toolkit)
+ */
+export interface ResponseToolkit {
+
+    /**
+     * A response symbol. When returned by a lifecycle method, the request lifecycle skips to the finalizing step
+     * without further interaction with the node response stream. It is the developer's responsibility to write
+     * and end the response directly via [request.raw.res](https://github.com/hapijs/hapi/blob/master/API.md#request.raw).
+     */
+    readonly abandon: Object;
+
+    /**
+     * A response symbol. When returned by a lifecycle method, the request lifecycle skips to the finalizing step after
+     * calling request.raw.res.end()) to close the the node response stream.
+     */
+    readonly close: Object;
+
+    /**
+     * A response symbol. Provides access to the route or server context set via the route [bind](https://github.com/hapijs/hapi/blob/master/API.md#route.options.bind)
+     * option or [server.bind()](https://github.com/hapijs/hapi/blob/master/API.md#server.bind()).
+     */
+    readonly context: any;
```

Comment:
Technically should be set based on the the presence of `bind` in options via generics/templates, this should be achievement.

Reply:
Yes, this is tricky since the server has a bind method as well as route, but I'll see if I can get it done.  I'm going to save that for a last commit in case it turns into a debacle.

I'd like to make plugins generic as well, though the array form of the register method is troublesome...